src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -376,13 +376,13 @@ public override async Task RazorServerReadyAsync(CancellationToken cancellationT
 
         private static bool SupportsCSharpCodeActions(JToken token)
         {
-            var serverCapabilities = token.ToObject<VSServerCapabilities>();
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
 
             var providesCodeActions = serverCapabilities?.CodeActionProvider?.Match(
                 boolValue => boolValue,
                 options => options != null) ?? false;
 
-            var resolvesCodeActions = serverCapabilities?.CodeActionsResolveProvider == true;
+            var resolvesCodeActions = serverCapabilities?.CodeActionProvider?.Second?.ResolveProvider == true;
```

Comment:
SumType.Second will throw an InvalidCastException if its value is bool. I would use Match again instead.